torchgeo/datasets/south_africa_crop_type.py

```diff
@@ -0,0 +1,248 @@
+    S2_bands = [
+        "B01",
+        "B02",
+        "B03",
+        "B04",
+        "B05",
+        "B06",
+        "B07",
+        "B08",
+        "B8A",
+        "B09",
+        "B11",
+        "B12",
+    ]
+    all_bands: list[str] = S1_bands + S2_bands
+    cmap = {
+        0: (0, 0, 0, 255),
+        1: (255, 211, 0, 255),
+        2: (255, 37, 37, 255),
+        3: (0, 168, 226, 255),
+        4: (255, 158, 9, 255),
+        5: (37, 111, 0, 255),
+        6: (255, 255, 0, 255),
+        8: (111, 166, 0, 255),
+        9: (0, 175, 73, 255),
+    }
+
+    def __init__(
+        self,
+        root: str = "",
+        crs: CRS = CRS.from_epsg(32634),
+        classes: list[int] = list(cmap.keys()),
+        bands: list[str] = all_bands,
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+    ) -> None:
+        """Initialize a new South Africa dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: coordinate reference system to be used
+            classes: crop type classes to be included
+            bands: the subset of bands to load
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert (
+            set(classes) <= self.cmap.keys()
+        ), f"Only the following classes are valid: {list(self.cmap.keys())}."
+        assert 0 in classes, "Classes must include the background class: 0"
+
+        self.root = root
+        self.classes = classes
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((len(self.classes), 4), dtype=torch.uint8)
+
+        super().__init__(paths=root, crs=crs, bands=bands, transforms=transforms)
+
+        # Map chosen classes to ordinal numbers, all others mapped to background class
+        for v, k in enumerate(self.classes):
+            self.ordinal_map[k] = v
+            self.ordinal_cmap[v] = torch.tensor(self.cmap[k])
+
+    def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data, label, and field ids at that index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        filepaths = cast(list[str], [hit.object for hit in hits])
+
+        if not filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        data_list: list[Tensor] = []
+        filename_regex = re.compile(self.filename_regex, re.VERBOSE)
+
+        sample_dates_s1: dict[str, str] = {}
+        sample_dates_s2: dict[str, str] = {}
+        for band in self.bands:
```

Comment:
Remind me again why we have to copy all of this instead of just calling `super().__getitem__(...)`?